src/layers/learning/fully_connected.cpp

```diff
@@ -823,12 +823,18 @@ void fully_connected_layer<T, L, D>::write_specific_proto(
 template <typename TensorDataType, data_layout T_layout, El::Device Dev>
 void fully_connected_layer<TensorDataType, T_layout, Dev>::fp_compute()
 {
+  if (!this->weights_values(0).Participating())
+    return;
```

Comment:
will we have to do this in every layer? shouldn't this just be in `class layer`/`class model`? If not, what makes FullyConnected a special case?

Reply:
I would prefer to make `Layer::forward_prop()` a "template method"-type driver function that can skip it entirely:
```
Layer::forward_prop() {
  if (not this->is_participating())
    return;
  // maybe some other common functionality (profiling annotations?)
  // finally, layer-specific bits via the "fp_compute" worker function they already implement:
  fp_compute();
  // maybe some common cleanup?
}
```
I'm evaluating whether this pattern actually works for the current set of layers in LBANN (in the worst case, I could make `forward_prop` a `virtual` function with a comment that it should only be overriden in exceptional cases where the developer is very aware of what they're doing, thereby shifting the onus onto them to replicate the common bits/participation bits as appropriate.

Reply:
I'm updating `data_type_layer::forward_prop` and `data_type_layer::back_prop_impl_` to try to figure this out.